### ISSUE TITLE
Callable interface cannot be extended with new props since 0.37

### ISSUE THREAD

**Issue by minedeljkovic** [facebook/flow#3084]
In flow 0.37, it is impossible to introduce new props on extended callable interface.
For each new prop, flow reports: `property b. Property not found in Object`

```js
declare interface A {
  (x: string): string;
}

declare interface B extends A {
  b: string;
}
```
[Try flow](https://flowtype.org/try/#0CYUwxgNghgTiAEBLAdgFxDAZlMCCC8A3gFDzwAUAHgFzwDOqMKA5gJS0NPLMDcxAvsWKhIsBCnRYcCAELwQldMmB14BEmQBGHRiz6CgA)

For simple (not callable) interface, this is not reported.

**Workaround** (thanks @philikon):
```js
declare interface A {
  // (x: string): string;
  $call: (x: string) => string
}

declare interface B extends A {
  b: string;
}

declare var a: A;
a('')
```

[Try flow](https://flowtype.org/try/#0CYUwxgNghgTiAEBLAdgFxDAZlMCCC8A3gFDzwD058AFAB4Bc8AzqjCgOYCUjLby7AblLwAJGCgQIjOj1YdO8ALwA+ZnP7EAvsWKhIsBCnRYcCAELwQtdMmBN4BEmQBGsvoK0690OPABusPBQjHhCUNQA5BGcQA)

Edit: workaround added

**Comment by philikon** [comment]
Another way this breaks down: `express`'s `Application` type is in fact a callable that mixes in `EventEmitter` and the `Application` prototypes. If I were to add a callable signature to in https://github.com/flowtype/flow-typed/blob/622c2ee76da078f51a05cbc90708d7475eca7103/definitions/npm/express_v4.x.x/flow_v0.28.x-/express_v4.x.x.js#L140-L160, Flow 0.37.x throws away the entire class definition and assumes `Application` is merely an `Object`.

Pretty please fix? :)

**Comment by philikon** [comment]
FWIW, the special `$call` property makes for a passable workaround, it seems. See https://github.com/flowtype/flow-typed/pull/533.

**Comment by samwgoldman** [comment]
Fixed since 0.55
